src/core/border_router/routing_manager.cpp

```diff
@@ -3110,6 +3161,172 @@ void RoutingManager::RsSender::HandleTimer(void)
     return;
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
+RoutingManager::PdPrefixManager::PdPrefixManager(Instance &aInstance)
+    : InstanceLocator(aInstance)
+    , mEnabled(false)
+    , mTimer(aInstance)
+{
+    mPrefix.Clear();
+}
+
+Error RoutingManager::PdPrefixManager::GetPrefixInfo(PrefixTableEntry &aInfo) const
+{
+    Error     error = kErrorNone;
+    TimeMilli now   = TimerMilli::GetNow();
+
+    VerifyOrExit(IsRunning() && HasPrefix(), error = kErrorNotFound);
+
+    aInfo.mPrefix              = mPrefix.GetPrefix();
+    aInfo.mValidLifetime       = mPrefix.GetValidLifetime();
+    aInfo.mPreferredLifetime   = mPrefix.GetPreferredLifetime();
+    aInfo.mMsecSinceLastUpdate = now - mPrefix.GetLastUpdateTime();
+
+exit:
+    return error;
+}
+
+void RoutingManager::PdPrefixManager::WithdrawPrefix(void)
+{
+    VerifyOrExit(HasPrefix());
+
+    LogInfo("Withdrew platform provided outdated prefix: %s", mPrefix.GetPrefix().ToString().AsCString());
+
+    mPrefix.Clear();
+    mTimer.Stop();
+
+    Get<RoutingManager>().ScheduleRoutingPolicyEvaluation(kImmediately);
+
+exit:
+    return;
+}
+
+void RoutingManager::PdPrefixManager::HandleTimer(void)
+{
+    if (mPrefix.IsDeprecated())
+    {
+        WithdrawPrefix();
+    }
```

Comment:
Can simplify? since timer is now only used to deprecate the prefix.
```suggestion
        WithdrawPrefix();
```